### PR TITLE
Add segment async header

### LINKED ISSUE #343: Async header for subscriber's segment pages
Add async header for subscriber's segment pages, i.e. the header shows a list name and a dropdown with the delete action for segment.

image

### PR DESCRIPTION
**Pull request by astupak**
Closes #343

### CODE REVIEW

File: packages/sendsay-api/modules/segments/__tests__/creators.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import * as creators from '../creators';
+
+describe('creators', () => {
+  const id = 'p123';
+
+  it('returns corrent request for get segment ', () => {
+    expect(creators.get(id)).toEqual({
+      action: 'group.get',
+      id: 'p123',
```
Comment (zouxuoz) [inline packages/sendsay-api/modules/segments/__tests__/creators.js]: use `id`

File: packages/sendsay-api/modules/segments/__tests__/creators.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    });
+  });
+
+  it('returns corrent request for destroy segment', () => {
+    expect(creators.destroy(id)).toEqual({
+      action: 'group.delete',
+      id: 'p123',
```
Comment (zouxuoz) [inline packages/sendsay-api/modules/segments/__tests__/creators.js]: use `id`

File: packages/sendsay-api/modules/segments/__tests__/parsers.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import * as parsers from '../parsers';
+
+describe('parsers', () => {
+  it('returns segment', () => {
+    const res = {
+      obj: {
+        reltype: '0',
```
Comment (zouxuoz) [inline packages/sendsay-api/modules/segments/__tests__/parsers.js]: you test parsing for `name` - remove unnecessary properties

File: packages/sendsay-api/modules/segments/__tests__/creators.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import * as creators from '../creators';
+
+describe('creators', () => {
+  const id = 'p123';
+
+  it('returns corrent request for get segment ', () => {
```
Comment (rosendi) [inline packages/sendsay-api/modules/segments/__tests__/creators.js]: Fix the gramma: `returns request for fetching segment`

File: src/routes/subscribers/routes/segments/index.js [hunk 5]
```diff
@@ -1,39 +1,34 @@
 import React, { PropTypes } from 'react';
-import { Match, Redirect } from 'react-router';
+import { Match } from 'react-router';
 import asyncContainer from 'shared/containers/asyncContainer';
+import injectorContainer from 'shared/containers/injectorContainer';
+import { saga, reducer } from './modules/segment';
```
Comment (zouxuoz) [inline src/routes/subscribers/routes/segments/index.js]: module should be named as `segments`

File: packages/sendsay-api/modules/segments/parsers.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import parseObject from 'sendsay-api/parseObject';
```
Comment (rosendi) [inline packages/sendsay-api/modules/segments/parsers.js]: Use relative import

File: src/routes/subscribers/routes/segments/routes/show/components/SegmentHeader.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    id: 'segment-header__subscribers',
+    defaultMessage: 'Просмотр подписчиков',
+  },
+
+  edit: {
+    id: 'segment-header__edit',
+    defaultMessage: 'Настройка условий',
+  },
+
+  destroy: {
```
Comment (zouxuoz) [inline src/routes/subscribers/routes/segments/routes/show/components/SegmentHeader.js]: unused

File: src/routes/subscribers/routes/segments/modules/segment/reducers.js [hunk 8]
```diff
@@ -0,0 +1,125 @@
+import { combineReducers } from 'redux';
```
Comment (rosendi) [inline src/routes/subscribers/routes/segments/modules/segment/reducers.js]: Rename folder to 'segments'

File: src/routes/subscribers/routes/segments/routes/show/components/SegmentHeader.js (hunk 7 above)
Comment (rosendi) [inline src/routes/subscribers/routes/segments/routes/show/components/SegmentHeader.js]: means destroy is unused

File: src/routes/subscribers/routes/segments/routes/show/components/SegmentLayout.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
+import React, { PropTypes } from 'react';
```
Comment (rosendi) [inline src/routes/subscribers/routes/segments/routes/show/components/SegmentLayout.js]: Remove segment id. Use `segment.id`.

File: src/routes/subscribers/routes/segments/routes/show/containers/SegmentContainer.js [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+import SegmentLayout from './../components/SegmentLayout';
+
+class SegmentContainer extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    children: PropTypes.node.isRequired,
+    segmentId: PropTypes.string.isRequired,
+    fetching: PropTypes.bool.isRequired,
+    fetched: PropTypes.bool.isRequired,
+    segment: PropTypes.shape({
```
Comment (rosendi) [inline src/routes/subscribers/routes/segments/routes/show/containers/SegmentContainer.js]: Move it to the bottom.

File: src/routes/subscribers/routes/segments/routes/show/index.js [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+    <DialogMatch id="segment__destroy" component={SegmentDestroyDialogContainer} />
+
+  </SegmentContainer>
+);
+
+
+SegmentRoutes.propTypes = {
+  pathname: PropTypes.string.isRequired,
+  pattern: PropTypes.string.isRequired,
+  params: PropTypes.shape({
```
Comment (radiansz) [inline src/routes/subscribers/routes/segments/routes/show/index.js]: Also isRequired

**Comment by rosendi** [comment]
Add test for the `segment` module.

File: src/routes/subscribers/routes/segments/routes/show/components/SegmentHeader.js [hunk 12]
```diff
@@ -40,11 +39,6 @@ const SegmentHeader = ({ intl, segment }) => (
         <SegmentDropdownContainer segmentId={segment.id} />,
       ]}
     />
-    <SectionNavigation>
-      <SectionNavigationLink to={`/subscribers/segments/${segment.id}/summary`} label={intl.formatMessage(messages.summary)} />
-      <SectionNavigationLink to={`/subscribers/segments/${segment.id}/subscribers`} label={intl.formatMessage(messages.subscribers)} />
-      <SectionNavigationLink to={`/subscribers/segments/${segment.id}/edit`} label={intl.formatMessage(messages.edit)} />
-    </SectionNavigation>
   </header>
```
Comment (radiansz) [inline src/routes/subscribers/routes/segments/routes/show/components/SegmentHeader.js]: Get rid of `header` tag

File: src/routes/subscribers/routes/segments/routes/show/components/SegmentNavigation.js [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+
+  edit: {
+    id: 'segment-header__edit',
+    defaultMessage: 'Настройка условий',
+  },
+
+});
+
+const SegmentNavigation = ({ intl, segment }) => (
+  <header>
```
Comment (radiansz) [inline src/routes/subscribers/routes/segments/routes/show/components/SegmentNavigation.js]: Get rid of `header`

**Comment by radiansz** [comment]
lgtm :rabbit:

**Comment by zouxuoz** [comment]
lgtm 🤒

**Comment by radiansz** [comment]
:tanabata_tree: lgtm

**Pull request merged**